public/app/features/explore/utils/decorators.ts

```diff
@@ -82,11 +88,14 @@ export const decorateWithGraphResult = (data: ExplorePanelData): ExplorePanelDat
 
 /**
  * This processing returns Observable because it uses Transformer internally which result type is also Observable.
- * In this case the transformer should return single result but it is possible that in the future it could return
+ * In this case the transformer should return single result, but it is possible that in the future it could return
  * multiple results and so this should be used with mergeMap or similar to unbox the internal observable.
  */
 export const decorateWithTableResult = (data: ExplorePanelData): Observable<ExplorePanelData> => {
-  if (data.tableFrames.length === 0) {
+  // Prometheus has a custom frame visualization alongside the table view, but they both handle the data the same
+  const tableFrames = data.tableFrames.length ? data.tableFrames : data.rawPrometheusFrames;
```

Comment:
Hmm... I don't think we can do merging tables and prometheus in single pass. We may need to do it for table and rawPrometheus separately because there might be cases where both `data.tableFrames` and `data.rawPrometheusFrames` have some values and we don't want to merge them together. `rawPrometheus` and `table` frames should be merged and visualized separately.

Reply:
I took a stab at this, can you take a look and let me know if that's what you were expecting?